### ISSUE TITLE
Whitespace handling support

### ISSUE THREAD

**Issue by fedorusov** [antlr/antlr4#2628]
Some languages require rare whitespace handling for disambiguation. For now only possible solution is to inject useless possible whitespaces everywhere what making grammar unreadable and error-prone.   
Possible antlr framework improvement is to add special constructs like \<enableHIDDEN\> \<disableHIDDEN\> (which manipulate HIDDEN channel handling at parse time) and place them when necessary.  
[There is already exist](https://stackoverflow.com/a/29115489/11630974) some solution with custom tokenStream, but its only application is to ensure if there is a whitespace. If you must choose appropriate alternative based on whitespace appeareance there is just no means to execute some code at prediction time.  
To sum up there can be brand new antlr concept with enabling HIDDEN channel when necessary.
